handler/src/main/java/io/netty/handler/ssl/OpenSslEngine.java

```diff
@@ -669,120 +650,144 @@ public synchronized SSLEngineResult unwrap(
+                            // we will produce an error on the next write attempt, which will trigger a SSL.clearError()
+                            // later.
+                            break;
+                        }
                     } else {
-                        // We were not able to write everything into the BIO so break the write loop as otherwise
-                        // we will produce an error on the next write attempt, which will trigger a SSL.clearError()
-                        // later.
+                        // BIO_write returned a negative or zero number, this means we could not complete the write
+                        // operation and should retry later.
+                        // We ignore BIO_* errors here as we use in memory BIO anyway and will do another SSL_* call
+                        // later on in which we will produce an exception in case of an error
+                        SSL.clearError();
                         break;
                     }
-                } else {
-                    // BIO_write returned a negative or zero number, this means we could not complete the write
-                    // operation and should retry later.
-                    // We ignore BIO_* errors here as we use in memory BIO anyway and will do another SSL_* call later
-                    // on in which we will produce an exception in case of an error
-                    SSL.clearError();
-                    break;
-                }
-            } while (srcsOffset < srcsEndOffset);
-        }
+                } while (srcsOffset < srcsEndOffset);
+            }
 
-        // Number of produced bytes
-        int bytesProduced = 0;
+            // Number of produced bytes
+            int bytesProduced = 0;
 
-        if (capacity > 0) {
-            // Write decrypted data to dsts buffers
-            int idx = dstsOffset;
-            while (idx < endOffset) {
-                ByteBuffer dst = dsts[idx];
-                if (!dst.hasRemaining()) {
-                    idx ++;
-                    continue;
-                }
+            if (capacity > 0) {
+                // Write decrypted data to dsts buffers
+                int idx = dstsOffset;
+                while (idx < endOffset) {
+                    ByteBuffer dst = dsts[idx];
+                    if (!dst.hasRemaining()) {
+                        idx++;
+                        continue;
+                    }
 
-                int bytesRead = readPlaintextData(dst);
+                    int bytesRead = readPlaintextData(dst);
 
-                // TODO: We may want to consider if we move this check and only do it in a less often called place at
-                // the price of not being 100% accurate, like for example when calling SSL.getError(...).
-                rejectRemoteInitiatedRenegation();
+                    // TODO: We may want to consider if we move this check and only do it in a less often called place
```

Comment:
![INFO](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-info.png) Complete the task associated to this TODO comment. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://garage.netty.io/sonarqube/coding_rules#rule_key=squid%3AS1135)